### DISCUSSION TITLE
How to append multiple params when using message keyboards?

### DISCUSSION THREAD

**Discussion by jialeee17** [def-studio/telegraph#545] [Q&A]
Hello, may I know how can I add more than 1 param when using the message keyboard with callback buttons?

This is not working...
`
Button::make('Delete')->action('delete')->param('id', '42')->param('extra_data', 123),
`

**Comment by fabio-ivona** [comment]
Hello @jialeee17 

could you paste the result of this?

`Button::make('Delete')->action('delete')->param('id', '42')->param('extra_data', 123)->toArray()`

**Reply by jialeee17** [reply]
The previous code changed a little bit, but the problem is still remain the same. Here's the updated code and the result:

`Button::make(__('telegram.yes'))->action('choose_account_deletion_confirmation')->param('login_id', $loginId)->param('answer', true)->toArray()` returns 
`array:2 [
  "text" => "Yes"
  "callback_data" => "action:choose_account_deletion_confirmation;login_id:22489753;answer:1"
]`

It didn't throw any error, the bot just didn't send the message when I use the "send()" method at the end. The entire code looks like:

`
                $this->chat->message(__('telegram.prompt.account_deletion_confirmation', ['account_label' => $account->label]))
                    ->keyboard(Keyboard::make()->buttons([
                        Button::make(__('telegram.yes'))
                            ->action('choose_account_deletion_confirmation')
                            ->param('login_id', $loginId)
                            ->param('answer', true)
                            ->width(0.5),
                        Button::make(__('telegram.no'))
                            ->action('choose_account_deletion_confirmation')
                            ->param('login_id', $loginId)
                            ->param('answer', false)
                            ->width(0.5),
                    ]))->send();
`

**Reply by jialeee17** [reply]
Updated: The problem seems to occur when either the action name or parameter name exceeds certain length.

**Reply by fabio-ivona** [reply]
there is a limit of 64bytes for parameters, that's the reason, I guess
